### PR TITLE
Support softmax return in softmax_with_cross_entropy

### PR DESCRIPTION
**Pull request by kuke**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_layers.py [hunk 1]
```diff
@@ -369,8 +369,10 @@ def test_softmax_with_cross_entropy(self):
         with program_guard(program):
             x = layers.data(name='x', shape=[16], dtype='float32')
             y = layers.data(name='label', shape=[1], dtype='int64')
-            loss = layers.softmax_with_cross_entropy(x, y)
+            loss, softmax = layers.softmax_with_cross_entropy(
```
Comment (panyx0718) [inline python/paddle/fluid/tests/unittests/test_layers.py:372]: also test return return_softmax=False?

Comment (kuke) [reply python/paddle/fluid/tests/unittests/test_layers.py:372]: Done

**Review by kuke — COMMENTED** [review]
THX

**Review by panyx0718 — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]
